core/src/main/java/hudson/model/ParametersDefinitionProperty.java

```diff
@@ -133,7 +133,9 @@ public void _doBuild(StaplerRequest req, StaplerResponse rsp) throws IOException
      * This method is supposed to be invoked from {@link ParameterizedJobMixIn#doBuild(StaplerRequest, StaplerResponse, TimeDuration)}.
      */
     public void _doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParameter TimeDuration delay) throws IOException, ServletException {
-        if (delay==null)    delay=new TimeDuration(getJob().getQuietPeriod());
+        if (delay==null)
+            // time set in UI is in seconds. Changing it into milliseconds
+            delay=new TimeDuration(getJob().getQuietPeriod() * 1000);
```

Comment:
It may not be super useful here, but when converting time units, I usually like to use the `TimeUnit` enum to convert. Makes it less necessary to comment on the units since it's obvious based on the code then.

Reply:
I'm not used to do it, but I like the idea.

Reply:
+1, there're a lot of examples in case you are looking for it in the core @fcojfernandez. https://github.com/jenkinsci/jenkins/search?q=TimeUnit&unscoped_q=TimeUnit

For instance, to convert 5 seconds into milliseconds: https://github.com/jenkinsci/jenkins/blob/d4737b2b60438ce0e3cb7a8a6f26675f7c880914/test/src/test/java/hudson/node_monitors/ClockMonitorDescriptorTest.java#L35